### PR TITLE
gdx-box2d: fix for #2750 - EndContact does not called

### PR DESCRIPTION
**Pull request by rinold**
This fixes 2 scenarios mentioned in #2750:
1. EndContact was not called when the fixture destroyed
2. EndContact was not called when the body set to inactive

Also Fixture user data was not set to null as done for Body user data when fixture destroyed

### CODE REVIEW

File: extensions/gdx-box2d/gdx-box2d/src/com/badlogic/gdx/physics/box2d/World.java [hunk 1]
```diff
@@ -341,6 +341,47 @@ public void destroyBody (Body body) {
...
 		world->SetContactListener(0);
 	*/
+	
+	/** Internal method for fixture destruction with notifying custom
+	 * contact listener
+	 * @param body
+	 * @param fixture */
+	void destroyFixture(Body body, Fixture fixture) {
+		jniDestroyFixture(addr, body.addr, fixture.addr);
+		fixture.setUserData(null);
```
Comment (rinold) [inline extensions/gdx-box2d/gdx-box2d/src/com/badlogic/gdx/physics/box2d/World.java]: Duplicate code, need to remove it.

**Comment by badlogic** [comment]
Looks good, @MobiDevelop can you sign this one off?

**Comment by MobiDevelop** [comment]
Does this need anything to be done in the GWT emulation?

**Comment by rinold** [comment]
@MobiDevelop, I wasn't able to reproduce the issue on GWT emulation - it works as expected.

**Comment by MobiDevelop** [comment]
Cool, thanks for verifying.

**Pull request merged**